questionpy_sdk/_package/_builder.py

```diff
@@ -86,12 +88,42 @@ def write_package(self) -> None:
             self._copy_source_files()
         self._run_build_hooks("post")
         self._handle_generated_static_files()
+        self._handle_migrations()
         self._write_manifest()
 
         validate_dist_structure(self._manifest, self._target.dist)
         validate_requested_lms_attributes(self._manifest)
         validate_package_name_and_description(self._manifest)
 
+    def _handle_migrations(self) -> None:
+        """Calculates the current state version of the package by counting the migrations.
+
+        This should be called after `questionpy` and other dependencies were installed.
+        """
+        sys.path.extend([str(self._target.dist / "dependencies" / "site-packages"), str(self._target.dist / "python")])
+
+        dont_write_bytecode = sys.dont_write_bytecode
+        sys.dont_write_bytecode = True
+
+        try:
+            migrations = get_migrations(self._manifest.namespace, self._manifest.short_name)
+        except MigrationError as e:
+            msg = "There was an error discovering the package migrations."
+            # TODO: Log (?) the full exception for further investigations.
+            raise PackageBuildError(msg) from e
+
+        self._manifest.state_version = len(migrations.package)
```

Comment:
Hier wäre es besser statt nach der Anzahl nach dem höchsten/max Integer zu sehen. Falls man doch alte Migrations mal löscht, sollte das nichts durcheinander bringen.

Falls die state_version schon in der qpy_config.yml explizit angegeben wurde, sollte es von dort übernommen werden. Falls der Wert dort und der automatisch ermittelte nicht übereinstimmen, könnte ja noch eine Warnung ausgegeben werden.

Reply:
Was genau meinst du mit "höchsten/max Integer"? Ich hatte das damals im Meeting angesprochen, dass wir es erlauben sollten, Migrations zu löschen (Digital Waste / Übersichtlichkeit)  - wir sind dann aber zum Schluss gekommen, dass wir das nicht wollen. Mit der jetztigen Implementierung könnten auch alle alten Migrations in einer Datei "versteckt" werden.